src/sort.jl

```diff
@@ -33,21 +53,83 @@ end
     length(a) <= 1 && return SVector{length(a),Int}(p)
 
     ordr = Perm(ord(lt, by, rev, order), a)
-    return SVector{length(a),Int}(_sort(p, alg, ordr))
+    return SVector{length(a),Int}(_sort(p, alg, isless, identity, nothing, ordr))
 end
 
+@inline _sort(a::StaticVector, alg, lt, by, rev, order) =
+    similar_type(a)(sort!(Base.copymutable(a); alg=alg, lt=lt, by=by, rev=rev, order=order))
+
+@inline _sort(a::StaticVector, alg::BitonicSortAlg, lt, by, rev, order) =
+    similar_type(a)(_sort(Tuple(a), alg, lt, by, rev, order))
+
+@inline _sort(a::NTuple, alg, lt, by, rev, order) =
+    sort!(Base.copymutable(a); alg=alg, lt=lt, by=by, rev=rev, order=order)
+
+@inline _sort(a::NTuple, ::BitonicSortAlg, lt, by, rev, order) =
+    _bitonic_sort(a, ord(lt, by, rev, order))
 
-@inline _sort(a::StaticVector, alg, order) =
-    similar_type(a)(sort!(Base.copymutable(a); alg=alg, order=order))
+# For better performance sorting floats under the isless relation, apply an order-preserving
+# bijection to sort them as integers.
+@inline function _sort(
+    a::NTuple{N, <:Union{Float16, Float32, Float64}}, ::BitonicSortAlg, lt, by, rev, order
+) where N
+    # Skip this special treatment when N = 2 to avoid a performance regression on AArch64.
+    N <= 2 && return _bitonic_sort(a, ord(lt, by, rev, order))
+    lt_rev = _simplify_order(lt, by, rev, order)
+    if lt_rev === nothing || lt_rev[1] !== isless
+        return _bitonic_sort(a, ord(lt, by, rev, order))
+    end
+    return _intfp.(_bitonic_sort(_fpint.(a), ord(isless, identity, lt_rev[2], Forward)))
+end
 
-@inline _sort(a::StaticVector, alg::BitonicSortAlg, order) =
-    similar_type(a)(_sort(Tuple(a), alg, order))
+# Given the order ord(lt, by, rev, order) on floats or integers, attempt to simplify it to
+# ord(_lt, identity, _rev, Forward), where _rev is a Bool and _lt is isless or <. If
+# successful, return (_lt, _rev). Otherwise, return `nothing`.
+@inline function _simplify_order(lt, by, rev::Union{Bool, Nothing}, order::Ordering)
+    (
+        any(Ref(lt) .=== (isless, <, >)) &&
```

Comment:
Good point. I've removed the function and switched to using mostly method specialization to do the same thing.

It's no longer relevant now that it's gone, but I agree that `any(Ref(f) .=== v)` is uglier than `f in v`. However, I found it necessary for full static evaluation when f is *not* the first element of `v`. For example, `(() -> any(Ref(-) .=== (+, -)))()` statically evaluates to true (`@code_llvm` is just`ret i8 1`), whereas `(() -> (-) in (+, -))()` compiles to some mess that evaluates at runtime to `true`. (Symbols have a [specialized `in` method](https://github.com/JuliaLang/julia/blob/v1.6.2/base/tuple.jl#L503) to avoid this.)

Reply:
Ah, I see, that is too bad. But it looks cleaner now.